src/grid/lebedev.py

```diff
@@ -73,145 +73,158 @@
-    ----------
-    sizes : array_like
-        Sequence of Lebedev grid sizes (e.g., number of points for each atomic shell).
-
-    Returns
-    -------
-    array_like
-        Sequence of the corresponding Lebedev grid degrees.
-
-    """
-    degrees = np.zeros(len(sizes), dtype=int)
-    for size in np.unique(sizes):
-        # get the degree corresponding to the given (unique) size
-        deg = _get_lebedev_size_and_degree(size=size)[0]
-        # set value of degree to corresponding to the given size equal to deg
-        degrees[np.where(sizes == size)] = deg
-    return degrees
-
-
-def _get_lebedev_size_and_degree(*, degree=None, size=None):
-    """Map the given degree and/or size to the degree and size of a supported Lebedev grid.
-
-    Parameters
-    ----------
-    degree : int, optional
-        Degree of Lebedev grid. If the Lebedev grid corresponding to the given degree is not
-        supported, the next largest degree is used.
-    size : int, optional
-        Number of Lebedev grid points. If the Lebedev grid corresponding to the given size is not
-        supported, the next largest size is used. If both degree and size are given, degree is
-        used for constructing the grid.
-
-    Returns
-    -------
-    int, int
-        Degree and size of a supported Lebedev grid (equal to or larger than the requested grid).
-
-    """
-    if degree and size:
-        warnings.warn(
-            "Both degree and size arguments are given, so only degree is used!",
-            RuntimeWarning,
-        )
-    if degree:
-        leb_degs = list(LEBEDEV_DEGREES.keys())
-        max_degree = max(leb_degs)
-        if degree < 0 or degree > max_degree:
+class AngularGrid(Grid):
+    """Unit spehrical integration grid."""
+
+    def __init__(
+        self, points=None, weights=None, *, degree=None, size=None, cache=True
+    ):
+        r"""Generate a Lebedev angular grid for the given degree and/or size.
+
+        Parameters
+        ----------
+        degree : int, optional
+            Degree of Lebedev grid. If the Lebedev grid corresponding to the given degree is not
+            supported, the next largest degree is used.
+        size : int, optional
+            Number of Lebedev grid points. If the Lebedev grid corresponding to the given size is
+            not supported, the next largest size is used. If both degree and size are given,
+            degree is used for constructing the grid.
+        cache : bool, optional
+            Enable to store loaded lebedeve grids in cache to avoid duplicate
+            file reading process. Default to `True`
+
+        Returns
+        -------
+        AngularGrid
+            An angular grid with Lebedev points and weights (including :math:`4\pi`) on
+            a unit sphere.
+
+        """
+        # construct grid from pts and wts given directly
+        if points is not None and weights is not None:
+            super().__init__(points, weights)
+            return
```

Comment:
Probably, not bad to print a statement only if `points` or `weights` are provided to inform the user that it is not being used.